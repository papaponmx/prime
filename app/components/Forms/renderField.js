import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { FieldWrapper } from './styles';

// TODO: Move this into Form/common inputs
const renderField = ({
  input,
  defaultValue,
  label,
  type,
  name,
  meta: { touched, error },
}) => (
  <FieldWrapper key={name}>
    <TextField
      {...input}
      label={label}
      type={type}
      id={name}
      name={name}
      error={touched && error}
      helperText={touched && error && error}
      defaultValue={defaultValue}
      InputLabelProps={{
        shrink: true,
      }}
    />
  </FieldWrapper>
);

// TODO: Pass texts as theme variables

renderField.propTypes = {
  input: PropTypes.any,
  label: PropTypes.string,
  type: PropTypes.string,
  meta: PropTypes.any,
  name: PropTypes.string,
  defaultValue: PropTypes.string,
};

export default renderField;
