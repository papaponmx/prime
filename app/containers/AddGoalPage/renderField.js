import React from 'react';
import PropTypes from 'prop-types';

const renderField = ({
  input,
  label,
  type,
  name,
  meta: { touched, error, warning },
}) => (
  <div>
    <label htmlFor={name}>{label}</label> {/* eslint-disable-line */}
    <div>
      <input {...input} placeholder={label} type={type} id={name} />
      {touched &&
        ((error && <span>{error}</span>) ||
          (warning && <span>{warning}</span>))}
    </div>
  </div>
);

renderField.propTypes = {
  input: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.any.isRequired,
  name: PropTypes.string.isRequired,
};

export default renderField;
