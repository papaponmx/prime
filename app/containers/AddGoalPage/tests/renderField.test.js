import React from 'react';
import { shallow } from 'enzyme';

import renderField from '../../../components/Forms/renderField';

describe('renderField()', () => {
  it('Should render field', () => {
    const wrapper = shallow(
      <renderField name="goal" type="text" label="Goal" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
