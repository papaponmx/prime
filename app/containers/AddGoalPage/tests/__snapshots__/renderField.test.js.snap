// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderField() Should render field 1`] = `
<renderField
  label="Goal"
  name="goal"
  type="text"
/>
`;
